src/Services/EncounterService.php

```diff
@@ -17,75 +17,72 @@
 namespace OpenEMR\Services;
 
 use OpenEMR\Common\Acl\AclMain;
+use OpenEMR\Common\Uuid\UuidRegistry;
 use OpenEMR\Services\FacilityService;
+use OpenEMR\Validators\EncounterValidator;
+use OpenEMR\Validators\ProcessingResult;
 use Particle\Validator\Validator;
 
-require_once(dirname(__FILE__) . "/../../library/forms.inc");
-require_once(dirname(__FILE__) . "/../../library/encounter.inc");
+require_once dirname(__FILE__) . "/../../library/forms.inc";
+require_once dirname(__FILE__) . "/../../library/encounter.inc";
 
-
-class EncounterService
+class EncounterService extends BaseService
 {
-
-  /**
-   * Default constructor.
-   */
+    private $encounterValidator;
+    private $uuidRegistery;
+    private const ENCOUNTER_TABLE = "form_encounter";
+    private const PATIENT_TABLE = "patient_data";
+
+    /**
+     * Default constructor.
+     */
     public function __construct()
     {
+        parent::__construct('form_encounter');
+        $this->uuidRegistery = new UuidRegistry(['table_name' => self::ENCOUNTER_TABLE]);
+        $this->uuidRegistery->createMissingUuids();
+        $this->encounterValidator = new EncounterValidator();
     }
 
-    public function validateEncounter($encounter)
+    /**
+     * Returns a single encounter record by encounter uuid and patient uuid.
+     * @param $puuid - The patient identifier of particular encounter
+     * @param $euuid - The encounter identifier used to lookup the encounter record.
+     * @return ProcessingResult which contains validation messages, internal error messages, and the data
+     * payload.
+     */
+    public function getEncounterForPatient($puuid, $euuid)
     {
-        $validator = new Validator();
-        $validator->required('date')->datetime('Y-m-d');
-        $validator->required('pc_catid');
-
 
-        return $validator->validate($encounter);
-    }
-    public function validateSoapNote($soapNote)
-    {
-        $validator = new Validator();
+        $processingResult = new ProcessingResult();
 
-        $validator->optional('subjective')->lengthBetween(2, 65535);
-        $validator->optional('objective')->lengthBetween(2, 65535);
-        $validator->optional('assessment')->lengthBetween(2, 65535);
-        $validator->optional('plan')->lengthBetween(2, 65535);
+        $isValidPatient = $this->encounterValidator->validateId('uuid', self::PATIENT_TABLE, $puuid, true);
+        if (is_array($isValidPatient)) {
+            $processingResult->setValidationMessages($isValidPatient);
+            return $processingResult;
+        }
+        $puuidBytes = UuidRegistry::uuidToBytes($puuid);
 
-        return $validator->validate($soapNote);
-    }
+        $pid = $this->getIdByUuid($puuidBytes, self::PATIENT_TABLE, "pid");
```

Comment:
nice line and function!
![2DV](https://user-images.githubusercontent.com/278968/83340466-6661df00-a28d-11ea-93c3-c130646eed65.gif)